src/main/java/com/minecolonies/coremod/entity/ai/statemachine/AITarget.java

```diff
@@ -78,7 +78,7 @@ public AITarget(@NotNull final IAIState predicateState, @Nullable final IAIState
      * @param state  the state it needs to be | null
      * @param action the action to apply
      */
-    public AITarget(@Nullable final IAIState state, @NotNull final Supplier<IAIState> action)
+    public AITarget(@NotNull final IAIState state, @NotNull final Supplier<IAIState> action)
```

Comment:
Please leave this as Nullable!
The API and JavaDoc explicitly state that this is allowed here!. This creates an Event Trigger.

Reply:
hm isnt the api/javadoc outated then? As of now we're using an additional Event types instead for those triggers, aka AIEventTarget

Reply:
No. It should allow it.

Reply:
But we've intentionally split away the Events from the AITarget to a seperate class, currently AITargets with no state arent even supported.

Reply:
Actually null classes can even cause issues now afaik.

Reply:
currently they're just not added to the statemachines transition list if it is a Transition without a state but not an event.